scripts/target.js

```diff
@@ -0,0 +1,281 @@
+const SESSION_COOKIE_NAME = 'sessionId';
+const SESSION_COOKIE_EXPIRATION_DAYS = 7;
+
+/**
+ * Escape a selector.
+ * @param selector
+ * @returns {string}
+ */
+function escapeSelector(selector) {
+  return selector.replaceAll(/#(\d)/g, '#\\3$1 ');
+}
+
+/**
+ * Generate a UUID.
+ * @returns {string}
+ */
+function uuid() {
+  return `sess-${Math.random()
+    .toString(36)
+    .substring(2, 9)}-${Date.now()
+    .toString(36)}`;
+}
+
+/**
+ * Get a cookie by name.
+ * @param name
+ * @returns {string}
+ */
+function getCookie(name) {
+  const cookies = document.cookie.split(';');
+  return cookies.find((cookie) => {
+    const [key, value] = cookie.split('=');
+    return key.trim() === name && value;
+  });
+}
+
+/**
+ * Set a cookie.
+ * @param name
+ * @param value
+ * @param days
+ */
+function setCookie(name, value, days) {
+  document.cookie = `${name}=${value}; max-age=${days * 24 * 60 * 60}; path=/`;
+}
+
+/**
+ * Get the session ID.
+ * @returns {string}
+ */
+function getSessionId() {
+  const existingSessionId = getCookie(SESSION_COOKIE_NAME);
+  if (existingSessionId) {
+    return existingSessionId;
+  }
+  const newSessionId = uuid();
+  setCookie(SESSION_COOKIE_NAME, newSessionId, SESSION_COOKIE_EXPIRATION_DAYS);
+  return newSessionId;
+}
+
+/**
+ * Get all offers from a response.
+ * @param data
+ * @returns {*[]}
+ */
+function getApplicableOffers(data) {
+  const offers = [];
+  const options = data.execute?.pageLoad?.options ?? [];
+  console.debug(`received ${options.length} options`); // eslint-disable-line no-console
+  options.forEach((option) => {
+    if (option.type === 'actions') {
+      option.content.forEach((content) => {
+        console.debug('processing content', content); // eslint-disable-line no-console
+        if (['setHtml', 'insertAfter', 'insertBefore'].includes(content.type)) {
```

Comment:
You could push either way, and just have the `default` in your `switch` handle the unknown commands. as it is the `default` would never be called